### DISCUSSION TITLE
Upgrading from v. 0.20.1

### DISCUSSION THREAD

**Discussion by TerryDanks** [Stellarium/stellarium#3106] [General]
Been using v.0.20.1 for many years. 
If I upgrade to the latest, will my settings be preserved. Specifically my customized treeline?

**Comment by gzotti** [comment]
When uninstalling, avoid answering YES to the question to remove all settings and data. Then settings will be preserved.

**Comment by TerryDanks** [comment]
Thank-you for the reply.
I have installed the latest version on another machine as a test. Despite having the same contents in the Landscapes folder as horizon.txt and landscape.ini, the tree horizon is not showing on the updated version. 

![StellariumLandscape](https://user-images.githubusercontent.com/128086724/225758756-7840c928-9adc-4a9e-ba52-46b25a710cf6.JPG)
![StellariumCapture2](https://user-images.githubusercontent.com/128086724/225758870-e5807b1f-b34e-42b1-ae9f-8777ea37e791.JPG)
Hope these images show? 
When I toggle the landscape button on/off the stars do wink on/off in the "hidden by trees" area. But the polygon line delineating the treeline does not show and all other objects, galaxies, constellation lines, etc. remain visible in the "hidden by trees" area. 
Suggestions?

**Reply by gzotti** [reply]
The "Geneva" polygonal landscape works today like it worked in 2016. Section 7.1.2 in the User Guide provides important information how to create polygonal landscapes. What else shall we describe?  If nothing else works, create a "spherical" landscape.

**Reply by worachate001** [reply]
@gzotti From your reply in #1296. It looks like polygonal_angle_rotatez must be non-zero. If it's 0, changing it to 0.00001 should fix the landscape.

**Reply by gzotti** [reply]
Good find :-) Yes, either the angles in the list should not contain 0, 90, 180, 270, or such tiny offset rotation should be used against the caveat mentioned in the Guide.

**Reply by TerryDanks** [reply]
Many thanks for this reference, Worachate.

My tree skyline is now showing in the latest version of Stellarium after I changed from zero to 0.00001 in 
the polygonal_angle_rotatez =  line of the landscape.ini file. :)
